File: front/src/app/models/MensajeChat.ts
export class MensajeChat {
	public idMensaje: string | null;
	public idCurso: number | null;
	public idClase: number | null;
	public idUsuario: number | undefined;

	public nombreCurso: string | null;
	public nombreClase: string | null;
	public nombreUsuario: string | null;

	public fotoCurso: string | null;
	public fotoUsuario: string | null;

	public respuesta: MensajeChat | null;
	public pregunta: number | null;

	public mensaje: string | null;

	public fecha: Date | null;

	constructor(_idMensaje: string | null, _idCurso: number | null, _idClase: number | null, _idUsuario: number | undefined, _nombreCurso: string | null, _nombreClase: string | null, _nombreUsuario: string | null, _fotoCurso: string | null, _fotoUsuario: string | null, _respuesta: MensajeChat | null, _pregunta: number | null, _mensaje: string | null, _fecha: Date | null) {
		this.idMensaje = _idMensaje;
		this.idCurso = _idCurso;
		this.idClase = _idClase;
		this.idUsuario = _idUsuario;
		this.nombreCurso = _nombreCurso;
		this.nombreClase = _nombreClase;
		this.nombreUsuario = _nombreUsuario;
		this.fotoCurso = _fotoCurso;
		this.fotoUsuario = _fotoUsuario;
		this.respuesta = _respuesta;
		this.pregunta = _pregunta;
		this.mensaje = _mensaje;
		this.fecha = _fecha;
	}
}

File: sov_school_back/back_chat/src/main/java/com/sovereingschool/back_chat/Models/MensajeChat.java
package com.sovereingschool.back_chat.Models;

import java.util.Date;

import org.springframework.data.mongodb.core.mapping.Document;

import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Document(collection = "messages_chat")
public class MensajeChat {
    @Id
    private String id;

    /** Curso ID */
    private Long idCurso;

    /** Clase ID */
    private Long idClase;

    /** Usuario ID de quien envía el mensaje */
    private Long idUsuario;

    /** ID del documento de respuesta (opcional) */
    private String respuesta;

    /** Segundo de video en la cuel se hizo la pregunta */
    private Integer momento;

    /** Texto del mensaje */
    private String mensaje;

    /** Fecha de envío del mensaje */
    private Date fecha;

}